sandberg_christy_ndvi.py

```diff
@@ -0,0 +1,700 @@
+full_ndvi_df = full_ndvi_df.set_index('date')
+
+# call dataframe at the end of the cell for autograding 
+full_ndvi_df
+
+
+# In[9]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points +=2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points +=2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points +=3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points +=3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# **NOTE:** The cell below removes the 'nan' values from the 'full_ndvi_df', keeping 
+# only the values needed to plot a continuous line for each NEON site.
+# 
+# Doing this will remove 15 'nan' rows from the dataframe.  
+# 
+# A sanity check looking at the nan values will fail, even though it seems to be 
+# looking for 15 nan values to be removed?
+# 
+# `(student_ndvi_df.mean_ndvi.isna().sum() == 15)`
+# 
+# *(Thanks to Lana, and her discussion post on this!)*
+
+# In[10]:
+
+
+# clean 'full_ndvi_df' for plot, removing 'nan' values by keeping only the 
+# rows with mean_ndvi > 0 
+ndvi_for_plot = full_ndvi_df[full_ndvi_df['mean_ndvi'] > 0]
+
+print(ndvi_for_plot)
+
+
+# In[11]:
+
+
+# Add only the plot code to this cell
+
+# This is the final figure of mean NDVI
+# for both sites across the year
+# with data cleaned to deal with clouds
+
+# YOUR CODE HERE
+
+# create plot showing mean ndvi for each site over time (include legend)
+colors = {"HARV": "firebrick",
+          "SJER": "teal"}
+
+fig, ax = plt.subplots()
+
+for label, df in ndvi_for_plot.groupby("site"):
```

Comment:
@csandberg303 Mitchell made a suggestion for line 589 that would require line 611 to have` full_ndvi_df` instead of `ndvi_for_plot` if you adopted that change

Reply:
@esgeo that's an excellent catch, Eric!  I added Mitchell's suggestion about how to drop the nan values, but I kept my original variable name so that it would match what I was using in the plot code.